lib/call-session.js

```diff
@@ -907,35 +907,38 @@ Duration=${payload.duration} `
         ...(req.has('X-Retain-Call-Sid') && {'X-Retain-Call-Sid': req.get('X-Retain-Call-Sid')}),
         ...(req.has('X-Account-Sid') && {'X-Account-Sid': req.get('X-Account-Sid')})
       };
-      const dlg = await this.srf.createUAC(referTo.uri, {localSdp: dlg.local.sdp, headers});
-      this.uas = dlg;
-      this.uas.other = this.uac;
-      this.uac.other = this.uas;
-      this.uas.on('modify', this._onReinvite.bind(this, this.uas));
-      this.uas.on('refer', this._onFeatureServerTransfer.bind(this, this.uas));
-      this.uas.on('destroy', () => {
+
+      const uac = await this.srf.createUAC(referTo.uri, {localSdp: dlg.local.sdp, headers});
+      this.uas = uac;
+      uac.type = 'uas';
```

Comment:
interesting, 'createUAC' does not do this?

Reply:
Are you referring to setting type?

We start with this scenario:

FS1 -> UAS (SBC) UAC -> End point

When the refer takes place we end up with:

FS2 <- UAC (SBC) UAC -> End point

Quite a few tests in the call-session check `(uac|uas).type` to handle things like INFO frames correctly, in my case `startCallRecording`. As the leg to FS2 is now a UAC we need to "correct" it so things are handled in the right way.

Hopefully that shouldn't cause a problem?